test/core/base/cache.js

```diff
@@ -9,35 +9,34 @@ describe('axe._cache', function () {
       assert.doesNotThrow(fn);
     });
 
-    // undefined is a valid result type of a potentially expensive operation in axe-core
     it('should set `undefined` without error', function () {
       function fn() {
         axe._cache.set('foo', undefined);
       }
       assert.doesNotThrow(fn);
     });
 
-    it('should throw TypeError for non-string keys', function () {
+    it('should throw for non-string keys', function () {
```

Comment:
Probably not a big deal, but IMO this should still be a `TypeError` (the error is due to the wrong type being used).

Reply:
Agreed, I would rather throw a more specific error. I changed this because I was asked to use the internal assert, which only throws errors.

@straker and @WilcoFiers  How would you feel about me adding an error type param to the axe assert helper so we can continue to assert the more specific type?

Reply:
I spoke with Wilco and Steve earlier today and discussed adding an error type to the internal assert helper to improve test quality. The rest of axe-core doesn't differentiate between error types, so they would rather not change the pattern for just the cache. I prefer the more specific error types, but it is probably a solution in search of a problem here.